### PR TITLE
py-sphinx-prompt: new package

### PR DESCRIPTION
**Pull request by LydDeb**
Add py-sphinx-prompt and its dependence py-poetry-plugin-tweak-dependencies-version

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
FYI. Confirmed the homepages and version sha256s.

File: var/spack/repos/builtin/packages/py-poetry-plugin-tweak-dependencies-version/package.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    homepage = "https://github.com/sbrunner/poetry-plugin-tweak-dependencies-version"
+    pypi = "poetry_plugin_tweak_dependencies_version/"
+    pypi += "poetry_plugin_tweak_dependencies_version-1.5.1.tar.gz"
+
+    maintainers("LydDeb")
+
+    version("1.5.1", sha256="4e0be2b8e23a04e542c5090deb5b6e191750ec45bace98ea8b55150844c6026b")
+
+    depends_on("py-poetry-core@1.0.0:", type="build")
+    depends_on("py-poetry-dynamic-versioning", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-poetry-plugin-tweak-dependencies-version/package.py:22]: Missing build/run deps:
```toml
[tool.poetry.dependencies]
python = ">=3.9,<3.12"
poetry = "==1.6.*"
```

File: var/spack/repos/builtin/packages/py-sphinx-prompt/package.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    maintainers("LydDeb")
+
+    version("1.8.0", sha256="47482f86fcec29662fdfd23e7c04ef03582714195d01f5d565403320084372ed")
+
+    depends_on("py-poetry-core@1.0.0:", type="build")
+    depends_on("py-poetry-plugin-tweak-dependencies-version", type="build")
+    depends_on("py-poetry-dynamic-versioning", type="build")
+    depends_on("py-sphinx", type=("build", "run"))
+    depends_on("py-pygments", type=("build", "run"))
+    depends_on("py-docutils", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sphinx-prompt/package.py]: Looks like these need to be exact versions:
```toml
[tool.poetry.dependencies]
python = ">=3.9,<4.0"
Sphinx = "7.2.5"
pygments = "2.16.1"
docutils = "0.20.1"
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-poetry-plugin-tweak-dependencies-version/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-poetry-plugin-tweak-dependencies-version/package.py:22]: You updated the wrong file

Comment (LydDeb) [reply var/spack/repos/builtin/packages/py-poetry-plugin-tweak-dependencies-version/package.py:22]: I add `py-poetry@1.6` in py-poetry-plugin-tweak-dependencies-version

File: var/spack/repos/builtin/packages/py-poetry-plugin-tweak-dependencies-version/package.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    """Poetry plugin used to tweak dependency versions."""
+
+    homepage = "https://github.com/sbrunner/poetry-plugin-tweak-dependencies-version"
+    pypi = "poetry_plugin_tweak_dependencies_version/"
+    pypi += "poetry_plugin_tweak_dependencies_version-1.5.1.tar.gz"
+
+    maintainers("LydDeb")
+
+    version("1.5.1", sha256="4e0be2b8e23a04e542c5090deb5b6e191750ec45bace98ea8b55150844c6026b")
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-poetry-plugin-tweak-dependencies-version/package.py:19]: Also needs:
```python
depends_on("python@3.9:3.11", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-sphinx-prompt/package.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    homepage = "https://github.com/sbrunner/sphinx-prompt"
+    pypi = "sphinx_prompt/sphinx_prompt-1.8.0.tar.gz"
+
+    maintainers("LydDeb")
+
+    version("1.8.0", sha256="47482f86fcec29662fdfd23e7c04ef03582714195d01f5d565403320084372ed")
+    version("1.1.0", sha256="3d9cf382b750291f73d1f6f1713c4af0557c30208af124cd3d8731e607a4febf")
+
+    depends_on("python@3.9:3.11", when="@1.8.0", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sphinx-prompt/package.py]:
```suggestion
    depends_on("python@3.9:3", when="@1.8.0", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-sphinx-prompt/package.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    depends_on("py-poetry@1.6", when="@1.8.0", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sphinx-prompt/package.py]: poetry is not a required dep

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**